src/compute/boolean.rs

```diff
@@ -133,3 +134,53 @@ pub fn is_not_null(input: &dyn Array) -> BooleanArray {
     };
     BooleanArray::from_data(DataType::Boolean, values, None)
 }
+
+/// Performs `AND` operation on an array and a scalar value. If either left or right value
+/// is null then the result is also null.
+/// # Example
+/// ```rust
+/// use arrow2::array::BooleanArray;
+/// use arrow2::compute::boolean::and_scalar;
+/// use arrow2::scalar::BooleanScalar;
+/// # fn main() {
+/// let array = BooleanArray::from_slice(&[false, false, true, true]);
+/// let scalar = BooleanScalar::new(Some(true));
+/// let result = and_scalar(&array, &scalar);
+/// assert_eq!(result, BooleanArray::from_slice(&[false, false, true, true]));
+/// # }
+/// ```
+pub fn and_scalar(array: &BooleanArray, scalar: &BooleanScalar) -> BooleanArray {
+    match scalar.value() {
+        Some(true) => array.clone(),
+        Some(false) => {
+            let values = Bitmap::from_trusted_len_iter(std::iter::repeat(false).take(array.len()));
+            BooleanArray::from_data(DataType::Boolean, values, array.validity().cloned())
+        }
+        None => BooleanArray::new_null(DataType::Boolean, array.len()),
+    }
+}
+
+/// Performs `OR` operation on an array and a scalar value. If either left or right value
+/// is null then the result is also null.
+/// # Example
+/// ```rust
+/// use arrow2::array::BooleanArray;
+/// use arrow2::compute::boolean::or_scalar;
+/// use arrow2::scalar::BooleanScalar;
+/// # fn main() {
+/// let array = BooleanArray::from_slice(&[false, false, true, true]);
+/// let scalar = BooleanScalar::new(Some(true));
+/// let result = or_scalar(&array, &scalar);
+/// assert_eq!(result, BooleanArray::from_slice(&[true, true, true, true]));
+/// # }
+/// ```
+pub fn or_scalar(array: &BooleanArray, scalar: &BooleanScalar) -> BooleanArray {
+    match scalar.value() {
+        Some(true) => {
+            let values = Bitmap::from_trusted_len_iter(std::iter::repeat(true).take(array.len()));
```

Comment:
```suggestion
            let values = Bitmap::from_len_zeroed(array.len());
```

Reply:
For this `or` version, I want to initialize a Bitmap with all `true` bits to implement `A | true = true`. Is it right?

Reply:
Oh, yes, you are right!

Unfortunately we do not have a specific method for that, but it is worth checking `MutableBitmap::extend_constant` here, since it sets 8 bits per write, still much faster than `from_trusted_len_iter` ^_^

Reply:
For context, the reason we have all these seemly similar APIs is that for a single API (like rust's `Vec::extend`) we need trait specialization in the stable channel. This is why we currently name every method differently.

Reply:
Thanks for instruction. I updated to use `MutableBitmap:: extend_constant`. Thanks.